### PR TITLE
Johannes/basic support 1

### PR DESCRIPTION
**Pull request by johannescoetzee**
# Notes
* Add support for conditional statements (`?:`)
* Add support for `foreach` loops. These are modeled as `for` nodes with 3 subtrees: the iterable, the local variable and the statement body. This is consistent with how these are handled in `c2cpg`.
* Add support for field accesses. TODO: Remove hack for resolving types during static member accesses.
# Testing
`sbt test`

### CODE REVIEW

File: src/test/scala/io/joern/javasrc2cpg/querying/ConditionalTests.scala [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+      |  }
+      |}
+      |""".stripMargin
+
+  "should parse ternary expression" in {
+    val List(ternaryExpr: Call) = cpg.call(Operators.conditional).l
+    val List(condition: Call, thenExpr: Literal, elseExpr: Call) = ternaryExpr.argument.l
+
+    condition.code shouldBe "x > 5"
+    condition.methodFullName shouldBe "<operator>.greaterThan"
```
Comment (ursachec) [inline src/test/scala/io/joern/javasrc2cpg/querying/ConditionalTests.scala:27]: just a nit, but you can use _Operators.greaterThan_ here

File: src/main/scala/io/joern/javasrc2cpg/passes/AstCreator.scala [hunk 2]
```diff
@@ -681,13 +693,86 @@ class AstCreator(filename: String, global: Global) {
...
+  }
+
+  def astForConditionalExpr(expr: ConditionalExpr, order: Int): Ast = {
+    val callNode = NewCall()
+      .name(Operators.conditional)
+      .methodFullName(Operators.conditional)
+      .dispatchType(DispatchTypes.STATIC_DISPATCH)
+      .code(expr.toString)
+      .argumentIndex(order)
+      .order(order)
```
Comment (ursachec) [inline src/main/scala/io/joern/javasrc2cpg/passes/AstCreator.scala:731]: you should add the line number here

File: src/test/scala/io/joern/javasrc2cpg/querying/ConditionalTests.scala (hunk 1 above)
Comment (johannescoetzee) [reply src/test/scala/io/joern/javasrc2cpg/querying/ConditionalTests.scala:27]: Good catch! The same goes for the addition below.

File: src/main/scala/io/joern/javasrc2cpg/passes/AstCreator.scala (hunk 2 above)
Comment (johannescoetzee) [reply src/main/scala/io/joern/javasrc2cpg/passes/AstCreator.scala:731]: Also a good point. I've updated this case now, but will do a more comprehensive check to see if line/col numbers exist and are what we expect them to for everything supported so far.

**Review by ursachec — APPROVED** [review]

**Pull request merged**